pkg/api/customization/authn/user_actions.go

```diff
@@ -136,3 +147,48 @@ func (h *Handler) setPassword(actionName string, action *types.Action, request *
 	request.WriteResponse(http.StatusOK, userData)
 	return nil
 }
+
+func (h *Handler) refreshAttributes(actionName string, action *types.Action, request *types.APIContext) error {
+	isAdmin, err := h.userIsAdmin(request)
+	if err != nil {
+		return err
+	}
+
+	if !isAdmin {
+		return errors.New("Not Allowed")
+	}
+
+	if request.ID != "" {
+		providerrefresh.TriggerUserRefresh(request.ID, true)
+	} else {
+		providerrefresh.TriggerAllUserRefresh()
+	}
+
+	request.WriteResponse(http.StatusOK, nil)
+	return nil
+}
+
+func (h *Handler) userIsAdmin(request *types.APIContext) (bool, error) {
+	store := request.Schema.Store
```

Comment:
change this so that it is not based on a rolename. use something like:
```
		return apiContext.AccessControl.CanDo(v3.UserGroupVersionKind.Group, v3.UserResource.Name, "create", apiContext, nil, apiContext.Schema) == nil
```
the idea being that people who can create users should be able to perform this action
you can work backwards form here for more examples: rancher/vendor/github.com/rancher/norman/types/server_types.go:97

Reply:
:+1: